dlr_vieten/explorer/templates/dlr_vieten_explorer_tolerance_factors.html

```diff
@@ -25,17 +26,202 @@
+                        var element = $('<div>'); element.addClass("input-group");
+                        var subelement = $('<span>');
+                        subelement.addClass('input-group-addon').html(c);
+                        element.append(subelement);
+                        var subelement = $('<input>');
+                        var first_key = Object.keys(data[c])[0];
+                        subelement.addClass('form-control').prop('type', 'text')
+                            .prop('placeholder', data[c][first_key]);
+                        element.append(subelement);
+                        var subelement = $('<span>');
+                        subelement.addClass('input-group-addon').html(first_key);
+                        element.append(subelement);
+                        wrapper.append(element);
+                        line.append(wrapper);
+                    }
+                };
+                $('#listTM').append(line);
+            });
+            $('#go1').on('click', function (e) {
+                var t = parseFloat($('#t').attr('placeholder'));
+                var r = parseFloat($('#r').attr('placeholder'));
+                var A_avg_Mg = 0; var A_avg_Ca = 0; var A_avg_Sr = 0; var A_avg_Ba = 0; var rO;
+                for (i = 0; i < table['rows'].length; i++) {
+                    var row = table['rows'][i];
+                    if (row['type'] == 'alkali earth') {
+                        var r_ion = parseFloat(row['rᵢₒₙ'].split()[0]);
+                        if (row['CN'] == '12') { if (row ['species'] == 'Mg')  { A_avg_Mg = r_ion * r; }}
+                        if (row['CN'] == '10') { if (row ['species'] == 'Mg')  { A_avg_Mg += r_ion * (1-r); }}
+                        if (row['CN'] == '12') { if (row ['species'] == 'Ca')  { A_avg_Ca = r_ion * r; }}
+                        if (row['CN'] == '10') { if (row ['species'] == 'Ca')  { A_avg_Ca += r_ion * (1-r); }}
+                        if (row['CN'] == '12') { if (row ['species'] == 'Sr')  { A_avg_Sr = r_ion * r; }}
+                        if (row['CN'] == '10') { if (row ['species'] == 'Sr')  { A_avg_Sr += r_ion * (1-r); }}
+                        if (row['CN'] == '12') { if (row ['species'] == 'Ba')  { A_avg_Ba = r_ion * r; }}
+                        if (row['CN'] == '10') { if (row ['species'] == 'Ba')  { A_avg_Ba += r_ion * (1-r); }}
+                    } else if (row['type'] == 'oxide ion') {
+                        rO = parseFloat(row['rᵢₒₙ'].split()[0]);
+                    }
+                }
+
+                //var nrTMs = $('#listTM').children.length;
+                var TMs = (1./nrTMs) // TODO  //
+                var TM3 = []; var TM4 = [];
+                $('#listTM').find('input').each(function (index) {
+                    var r_ion = parseFloat($(this).attr('placeholder').split()[0]);
+                    if (1) { TM4.push(r_ion); } //index%2 removed for now
```

Comment:
I guess `index%2` should be put back in again to fill TM4 and TM3!?

Reply:
I have a different solution for this now, will commit this soon